osbuild/pipeline.py

```diff
@@ -79,8 +63,13 @@ def run(self, tree, runner, build_tree, interactive=False, check=True, libdir=No
                     binds=[f"{tree}:/run/osbuild/tree"],
                     stdin=subprocess.DEVNULL,
                 )
-                if check and r.returncode != 0:
-                    raise StageFailed(self.name, r.returncode, api.output)
+                return {
+                    "name": self.name,
+                    "id": self.id,
+                    "options": self.options,
+                    "success": r.returncode == 0,
+                    "output": api.output
```

Comment:
The output itself contains json as well, right? And worse, probably some `"` chars that will totally confuse the json parser on the other side. Maybe we should `base64` encode the output data?

Reply:
This output only contains the logs right now, no json. (Even if it did, python's json library escapes everything properly.)

Once it does contain json, I'd prefer parsing it and inserting the object directly into the overall result.

Reply:
Yeah, I realized that today in the morning while still being in bed. It was late yesterday and I should have been in bed instead of commenting. All good, sorry ;)

Reply:
:laughing: no worries at all!